rcongui/src/components/PlayerInfo/PlayerInfo.js

```diff
@@ -0,0 +1,267 @@
+                        return true
+                })
+                if (temp !== undefined) {
+                    this.setState({temp: true})
+                }
+                if (perma !== undefined) {
+                    this.setState({perma: true})
+                }
+            })
+            .catch(handle_http_errors);
+    }
+
+    /**
+     * fetch Player data
+     * @param steamId64
+     */
+    #fetchPlayer = (steamId64) => {
+        get(`player?steam_id_64=${steamId64}`)
+            .then((response) => showResponse(response, "get_user", false))
+            .then((data) => {
+                this.setState({
+                    created: data.result.created,
+                    names: data.result.names,
+                    sessions: data.result.sessions,
+                    sessions_count: data.result.sessions_count,
+                    total_playtime_seconds: data.result.total_playtime_seconds,
+                    current_playtime_seconds: data.result.current_playtime_seconds,
+                    received_actions: data.result.received_actions,
+                    penalty_count: data.result.penalty_count,
+                    blacklist: data.result.blacklist,
+                    flags: data.result.flags,
+                    watchlist: data.result.watchlist,
+                    steaminfo: data.result.steaminfo,
+                    loaded: true
+                })
+
+            })
+            .catch(handle_http_errors);
+    }
+
+    componentWillReceiveProps(nextProps){
+        const { steamId64 } = nextProps.match.params
+        if(steamId64 !== undefined){
+            this.#fetchPlayer(steamId64)
+            this.#fetchPlayerBan(steamId64)
+        }
+    }
+
+    componentDidMount() {
+        const { steamId64 } = this.props.match.params
+        if(steamId64 !== undefined){
+            this.#fetchPlayer(steamId64)
+            this.#fetchPlayerBan(steamId64)
+        }
+    }
+
+    render() {
+        return (
+            <div className="player-info">
+                <div>
+                    {(this.state.loaded) ?
+                        <div className="player">
+                            <div className="player-col-1">
+                                <div className="avatar">
+                                    {
+                                        (this.state.steaminfo?.profile?.avatarfull !== undefined) ?
+                                            <img src={this.state.steaminfo?.profile?.avatarfull} alt="player avatar"/>
+                                            :
+                                            <span className="letter">{this.state.names[0]?.name[0].toUpperCase()}</span>
+                                    }
+
+                                </div>
+                                <div>
+                                    <h3>Last connection</h3>
+                                    <div>{moment(this.state.sessions[0]?.end).format("ddd Do MMM HH:mm:ss")}</div>
```

Comment:
`?.` nice, I did not know this operator existed.

Reply:
this one is so usefull